liquidity_pool/src/liquidity.rs

```diff
@@ -47,6 +47,7 @@ pub trait LiquidityModule:
             &amount,
             &[],
         );
+        EsdtTokenPayment::new(self.lend_token().get(), new_nonce, amount)
```

Comment:
You already have `lend_token_id` as variable, so use that instead of reading it from storage again.